BurnOutSharp/ProtectionType/Tages.cs

```diff
@@ -20,110 +20,190 @@ public string CheckContents(string file, byte[] fileContent, bool includeDebug,
+            /* 
+                Expected files to contain "TagesSetup"/"Application TagesSetup":
+                DrvSetup.exe
+                DrvSetup_x64.exe
+                TagesSetup.exe
+                TagesSetup_x64.exe
+             
+                Expected files to contain "Tages activation client"/"T@GES":
+                TagesClient.exe
+                (There is generally a TagesClient.dat accompanying the TagesClient.exe.)
+            */
+            string name = Utilities.GetFileDescription(pex);
+            if (!string.IsNullOrWhiteSpace(name) && name.StartsWith("TagesSetup", StringComparison.OrdinalIgnoreCase))
+                return $"TAGES Driver Setup {GetVersion(pex)}";
+            else if (!string.IsNullOrWhiteSpace(name) && name.StartsWith("Tagès activation client", StringComparison.OrdinalIgnoreCase))
+                return $"TAGES Activation Client {GetVersion(pex)}";
+
+            name = Utilities.GetProductName(pex);
+            if (!string.IsNullOrWhiteSpace(name) && name.StartsWith("Application TagesSetup", StringComparison.OrdinalIgnoreCase))
+                return $"TAGES Driver Setup {GetVersion(pex)}";
+            else if (!string.IsNullOrWhiteSpace(name) && name.StartsWith("T@GES", StringComparison.OrdinalIgnoreCase))
+                return $"TAGES Activation Client {GetVersion(pex)}";
 
             // TODO: Obtain a sample to find where this string is in a typical executable
-            if (includeDebug)
+            var contentMatchSets = new List<ContentMatchSet>
             {
-                var contentMatchSets = new List<ContentMatchSet>
+                // protected-tages-runtime.exe
+                new ContentMatchSet(new byte?[]
                 {
-                    // protected-tages-runtime.exe
-                    new ContentMatchSet(new byte?[]
-                    {
-                        0x70, 0x72, 0x6F, 0x74, 0x65, 0x63, 0x74, 0x65,
-                        0x64, 0x2D, 0x74, 0x61, 0x67, 0x65, 0x73, 0x2D,
-                        0x72, 0x75, 0x6E, 0x74, 0x69, 0x6D, 0x65, 0x2E,
-                        0x65, 0x78, 0x65
-                    }, Utilities.GetFileVersion, "TAGES"),
-
-                    // (char)0xE8 + u + (char)0x00 + (char)0x00 + (char)0x00 + (char)0xE8
-                    new ContentMatchSet(new byte?[] { 0xE8, 0x75, 0x00, 0x00, 0x00, 0xE8 }, GetVersion, "TAGES"),
-                };
-
-                return MatchUtil.GetFirstMatch(file, fileContent, contentMatchSets, includeDebug);
-            }
+                    0x70, 0x72, 0x6F, 0x74, 0x65, 0x63, 0x74, 0x65,
+                    0x64, 0x2D, 0x74, 0x61, 0x67, 0x65, 0x73, 0x2D,
+                    0x72, 0x75, 0x6E, 0x74, 0x69, 0x6D, 0x65, 0x2E,
+                    0x65, 0x78, 0x65
+                }, Utilities.GetFileVersion, "TAGES [DEBUG]"),
 
-            return null;
+                // This check seems to currently be broken, as files that appear to have this string aren't being detected.
```

Comment:
Are they hitting the `protected-tages-runtime.exe` check instead? The order of the checks matter here.

Reply:
None of these checks are being hit atm, and I don't know of any files at this moment that should ever hit this check.